src/uct/tcp/tcp_ep.c

```diff
@@ -1452,6 +1454,14 @@ uct_tcp_ep_am_prepare(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep,
     *hdr          = ep->tx.buf;
     (*hdr)->am_id = am_id;
 
+    ++ep->tx.sn;
+    if (ep->tx.sn == ep->last_acked_sn) {
+        /* If the TX sequence number is now the same as the last acked sequence
+         * number, ensure that they are different to request ACK through PUT in
+         * TCP ep flush operation */
+        --ep->last_acked_sn;
+    }
```

Comment:
can we do it for put operations only?

Reply:
unfortunately, no
if `ep->tx.sn` is wrapped and now it is  equal to `ep->last_acked_sn`, flush will return OK immediately, but we don't want it